allennlp/common/from_params.py

```diff
@@ -171,10 +181,17 @@ def create_kwargs(
         # it will have an __origin__ field indicating `typing.Dict`
         # and an __args__ field indicating `(str, int)`. We capture both.
         annotation = remove_optional(param.annotation)
-        kwargs[param_name] = construct_arg(
+
+        constructed_arg = pop_and_construct_arg(
             cls.__name__, param_name, annotation, param.default, params, **extras
         )
 
+        # If we just ended up constructing the default value for the parameter, we can just omit it.
+        # Leaving it in can cause issues with **kwargs in some corner cases, where you might end up
```

Comment:
Could you give an example of such a corner case? I can't think of how this could occur.

Reply:
Added an example to the comment.  If you do something like `super().__init__(lazy=True, **kwargs)` in a dataset reader, without this `if` you'd end up with both `lazy=True` (from your first argument) and `lazy=False` (inside of **kwargs, because we constructed the default value) in `DatasetReader.__init__`.